### PR TITLE
New package: libquo

### PR DESCRIPTION
**Pull request by KineticTheory**
QUO is a high-level, easy to use programming interface tailored specifically for MPI/MPI+X codes that may benefit from evolving process binding policies during their execution. QUO allows for arbitrary process binding policies to be enacted and reverted during the execution of an MPI/MPI+X application as different computational phases are entered and exited, respectively.

https://github.com/losalamos/libquo

### CODE REVIEW

File: var/spack/repos/builtin/packages/libquo/package.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from spack import *
+import os
+
+
+class Libquo(Package):
+    """A high-level, easy to use programming interface tailored specifically
+for MPI/MPI+X codes that may benefit from evolving process binding policies
+during their execution. QUO allows for arbitrary process binding policies to
+be enacted and reverted during the execution of an MPI/MPI+X application as
+different computational phases are entered and exited, respectively."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libquo/package.py]: This should probably all be indented by 4 spaces.

File: var/spack/repos/builtin/packages/libquo/package.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    version('1.2.9', 'ca82ab33f13e2b89983f81e7c02e98c2')
+
+    depends_on('mpi')
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+
+    def install(self, spec, prefix):
+        autoreconf = which("autoreconf")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libquo/package.py]: This is unnecessary. You should just be able to run `autoreconf(*autoreconf_options)` as the `autoreconf()` function is provided by the dependency on `autoconf`. See `setup_dependent_package()` in the autoconf package for details.

File: var/spack/repos/builtin/packages/libquo/package.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        autoreconf_options = [
+            '--install',
+            '--verbose',
+            '--force',
+            '-I', 'config',
+            '-I', os.path.join(spec['automake'].prefix,
+                               'share', 'aclocal'),
+            '-I', os.path.join(spec['libtool'].prefix,
+                               'share', 'aclocal'),
+            ]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libquo/package.py]: Indented 4 spaces too far.

File: var/spack/repos/builtin/packages/libquo/package.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        autoreconf(*autoreconf_options)
+
+        configure_options = [
+            '--prefix={0}'.format(prefix),
+            'CC=%s' % join_path(spec['mpi'].prefix.bin, "mpicc"),
+            'FC=%s' % join_path(spec['mpi'].prefix.bin, "mpif90") ]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libquo/package.py]: Can you move that bracket onto the last line? Flake8 doesn't like it like that.

**Comment by KineticTheory** [comment]
@adamjstewart Thanks for the comments.  I'll clean those things up and do some tests before updating this PR.  The `run-flake8-tests` script isn't working for me.  Any idea what this error means?

```
Traceback (most recent call last):
  File: "/scratch/vendors/python-2.7.12/bin/flake8", line 5, in <module>
    from pkg_resources import load_entry_point
ImportError: No module named pkg_resources
```

**Comment by adamjstewart** [comment]
@KineticTheory That error message means that it can't find setuptools. Unlike most other packages, py-flake8 actually requires py-setuptools as a runtime dependency. Add it to your PYTHONPATH or activate the py-setuptools extension.

**Comment by KineticTheory** [comment]
@adamjstewart BTW - activating py-setuptools fixed the flake8 issue.  Thanks for the tip.

**Comment by adamjstewart** [comment]
I wonder if there is a way I could check for that and print a more useful error message...

**Comment by adamjstewart** [comment]
I could add a test with `python -c "import setuptools"` to the `check_dependencies` script, but flake8 isn't built against the system python so I'm not sure if that would work. I would have to find flake8, extract the shebang, and use that Python. Hmm...

**Comment by adamjstewart** [comment]
Great. The flake8 package is missing a ton of dependencies anyway. Guess I'll kill two birds with one stone!

**Pull request merged**